### PR TITLE
Docs: Add warning about backend APIs not having auth by default

### PR DESCRIPTION
**Pull request by VladMasarik**
## Hey, I just made a Pull Request!
For me the realization that backstage was unauthenticated by default, and thus anyone can access its information, was very surprising and undesired. I could be the only one, but I would very much welcome any warning about this when reading a security section. AFAIK no such section exists so I put the warning into at least the authentication section.

Also, while reading the docs about cookie authentication, I thought that the "Asset requests" was rather confusing, as at first I was not sure what that is, and maybe "Frontend" would be more generic, but rather easier to understand.

I am not sure which package this should belong to, assuming I need to add a change set. Also, this is just a more concrete suggestion compared to an issue, so no hard feelings if this gets closed, and any feedback is welcome!

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Makes sense to lift this up a bit more. There's already a warning down at https://backstage.io/docs/auth/#sign-in-configuration, do you think it makes sense to remove? Might make sense to use some of that content too.

File: docs/auth/index.md
Comment (VladMasarik) [reply docs/auth/index.md:13]: Certainly, but I would like to get more clarification on these 3 texts.
1. Warning in https://backstage.io/docs/overview/threat-model/#integrator-responsibilities
2. Note in https://backstage.io/docs/auth/#sign-in-configuration
3. This PR note

My main goal with the suggested note is to make it explicit that the API exposed, and I would not consider the first and second texts to be related to my suggestion. However, if we want to make it clear that the API is exposed and the main idea is for the "Integrator" to ensure security, then I think I should actually edit the second note no? 

Would the following be more preferred? Include an explicit warning that the API is unauthenticated in the second note, with also a link to the first warning, instead of my additional note.

Also, this might not be the best place to discuss this, as I don't know the whole architectural decisions of Backstage, but are you sure not including link to the cookie tutorial is desired? I am somewhat confused as to why Backstage provides so many features for Identity management, authentication, and secure communication between plugins, but doesn't provide "the whole package". If you already strongly suggest to run backstage behind an Oauth proxy or some auth layer, wouldn't it be easier for backstage devs to drop these features, and shift all work towards the "owner"? Authorization included.

File: .changeset/slimy-spiders-pull.md
Comment (awanlin) [inline .changeset/slimy-spiders-pull.md:5]: You don't need a changeset for this and I think having this show up in the `CHANGELOG` for this package could cause confusion

Comment (VladMasarik) [reply .changeset/slimy-spiders-pull.md]: I have removed the changeset

File: docs/auth/index.md
Comment (Rugvip) [reply docs/auth/index.md:13]: I'm thinking we get rid of the second note and put it all in this top one instead.

We want to provide full auth out of the box, but it's WIP. We don't have the bandwidth to pick it up yet but we're hoping to have it done this year. Bit more info over here: #15999

Comment (awanlin) [reply docs/auth/index.md:13]: @VladMasarik, will you be able to make the above change, please? Once that's done we can get this approved and merged in

Comment (VladMasarik) [reply docs/auth/index.md]: @awanlin @Rugvip sorry for the hold up, I have merged the two texts, and added a link to the suggested "threat model". I am not sure it is the best, so I am very open to any suggestions.

**Review by awanlin — APPROVED** [review]

File: docs/auth/index.md
Comment (awanlin) [inline docs/auth/index.md:18]: Looks like this link is failing the build: `Unable to reach ../overview/threat-model from root or microsite/static/, linked from docs/auth/index.md`

**Comment by awanlin** [comment]
@Rugvip, this looks good to me, can you approve and merge if you feel the same, please?

File: docs/auth/index.md
Comment (VladMasarik) [reply docs/auth/index.md]: right, missed the file extension

**Review by tudi2d — APPROVED** [review]

File: docs/permissions/getting-started.md
Comment (Rugvip) [inline docs/permissions/getting-started.md:39]:
```suggestion
Asset requests initiated by the browser will not include a token in the `Authorization` header. If these requests check authorization through the permission framework, as done in plugins like TechDocs, then you'll need to set up cookie-based authentication. Refer to the ["Authenticate API requests"](https://github.com/backstage/backstage/blob/master/contrib/docs/tutorials/authenticate-api-requests.md) tutorial for a demonstration on how to implement this behavior.
```

Keeping this as is, it's specifically talking about requests for images, CSS files, etc. Requests in general will often have the auth header set.

**Review by Rugvip — APPROVED** [review]
Thank you! Looks good, just a tiny tweak and you'll :shipit:

**Pull request merged**
